analysers/experiments/IO.py

```diff
@@ -0,0 +1,148 @@
+from pyspark import SparkConf
+
+# Takes arguments: Spark master, Cassandra host, Minio host, path of the files
+sparkMaster = sys.argv[1]
+cassandraHost = sys.argv[2]
+trialID = sys.argv[3]
+experimentID = trialID.split("_")[0]
+cassandraKeyspace = "benchflow"
+srcTable = "trial_io"
+destTable = "exp_io"
+
+# Set configuration for spark context
+conf = SparkConf() \
+    .setAppName("IO analyser") \
+    .setMaster(sparkMaster) \
+    .set("spark.cassandra.connection.host", cassandraHost)
+sc = CassandraSparkContext(conf=conf)
+
+# TODO: Use Spark for all computations
+
+def f1(r):
+    if r['reads'] == None:
+        return (0, 1)
+    else:
+        return (r['reads'], 1)
+    
+def f2(r):
+    if r['writes'] == None:
+        return (0, 1)
+    else:
+        return (r['writes'], 1)
+
+def f3(r):
+    if r['total'] == None:
+        return (0, 1)
+    else:
+        return (r['total'], 1)
+    
+data = sc.cassandraTable(cassandraKeyspace, srcTable)\
+        .select("device") \
+        .where("experiment_id=? AND container_id=?", trialID, experimentID, containerID) \
+        .collect()
+
+devices = {}
+for e in data:
+    dev = e["device"]
+    devices[dev] = 0
+    
+queries = []
+
+def computeMetrics(op, dev):
+    dataRDD = sc.cassandraTable(cassandraKeyspace, srcTable) \
+            .select("device", op) \
+            .where("experiment_id=? AND container_id=?", experimentID, containerID) \
+            .filter(lambda a: a["device"] == d) \
+            .map(f) \
+            .cache()
+    
+    data = dataRDD.reduceByKey(lambda a, b: a + b) \
+            .map(lambda x: (x[1], x[0])) \
+            .sortByKey(0, 1) \
+            .collect()
+        
+    mode = list()
+    highestCount = data[0][0]        
+    for d in data:
+        if d[0] == highestCount:
+            mode.append(d[1])
+        else:
+            break
+    
+    data = dataRDD.sortByKey(0, 1) \
+            .map(lambda x: x[0]) \
+            .collect()
+     
+    #mode = data[0]
+    dataMin = data[-1]
+    dataMax = data[0]
+    
+    #def computeMedian(d):
+    #    leng = len(d)
+    #    if leng %2 == 1:
+    #        medianIndex = ((leng+1)/2)-1
+    #        return (d[medianIndex], medianIndex, "odd")
+    #    else:
+    #        medianIndex = len(d)/2
+    #        lower = d[leng/2-1]
+    #        upper = d[leng/2]
+    #        return ((float(lower + upper)) / 2.0, medianIndex, "even")
+    
+    dataLength = len(data)
+    #m = computeMedian(data)
+    #median = m[0]
+    #if m[2] == "odd":
+    #    q3 = computeMedian(data[0:m[1]])[0]
+    #else:
+    #    q3 = computeMedian(data[0:m[1]-1])[0]
+    #q2 = m[0]
+    #if m[2] == "even":
+    #    q1 = computeMedian(data[m[1]:])[0]
+    #else:
+    #    q1 = computeMedian(data[m[1]+1:])[0]
+    median = np.percentile(data, 50).item()
+    q1 = np.percentile(data, 25).item()
+    q2 = median
+    q3 = np.percentile(data, 75).item()
+    p95 = np.percentile(data, 95).item()
+          
```

Comment:
What kind of precision to we have for `double` data in the following operations?